qutebrowser/browser/shared.py

```diff
@@ -355,23 +355,60 @@ def choose_file(multiple: bool) -> List[str]:
         A list of selected file paths, or empty list if no file is selected.
         If multiple is False, the return value will have at most 1 item.
     """
-    handle = tempfile.NamedTemporaryFile(prefix='qutebrowser-fileselect-', delete=False)
-    handle.close()
-    tmpfilename = handle.name
-    with utils.cleanup_file(tmpfilename):
-        if multiple:
-            command = config.val.fileselect.multiple_files.command
-        else:
-            command = config.val.fileselect.single_file.command
+    if multiple:
+        command = config.val.fileselect.multiple_files.command
+    else:
+        command = config.val.fileselect.single_file.command
+    use_tmp_file = '{}' in command
```

Comment:
I think it's fine to rely on `{}` as a marker to say "pass a filename here" instead of using stdout. Are there any other approaches you can think of? The only thing I can think of are `fileselect.{multiple_files,single_file}.use_stdout` booleans, but that seems kind of overkill to me for something we can easily "detect" in a more or less intuitive way.

@user202729 Do you have any opinion on this?

Reply:
I had nothing concrete in mind but wanted to see what you thought of it. So let's keep `{}` as the marker :+1:

Reply:
At the moment `bash -c "file-picker-command > {}"` already work,  but detecting `> {}` at the end of the command-line (or run it in a shell automatically) could be even worse (there might be some weird file picker that takes `> filename` to specify output file path).

Besides, currently `replace('{}', tmpfilename)` is used anyway, so ~~the behavior is consistent~~.

(the behavior is definitely not consistent. The consistent way must be `any('{}' in arg for arg in command[1:])`.)

(I just notice that there's no easy way to escape literal `{}`.)

Reply:
For the find case, if I recalled correctly there's a `-I` flag that allow you to specify a different place holder.